Lib/selectors.py

```diff
@@ -547,23 +547,21 @@ def select(self, timeout=None):
             # If max_ev is 0, kqueue will ignore the timeout. For consistent
             # behavior with the other selector classes, we prevent that here
             # (using max). See https://bugs.python.org/issue29255
-            max_ev = max(len(self._fd_to_key), 1)
+            max_ev = len(self._fd_to_key) or 1
             ready = []
             try:
                 kev_list = self._selector.control(None, max_ev, timeout)
             except InterruptedError:
                 return ready
+
+            fd_to_key = self._fd_to_key
```

Comment:
```suggestion
            fd_to_key_get = self._fd_to_key.get
```
Since we only use the get method, we can use that one